cardano-node/src/Cardano/Common/Parsers.hs

```diff
@@ -49,33 +57,69 @@ parseNodeId desc =
     )
 
 -- | Flag parser, that returns its argument on success.
-fl :: a -> String -> String -> Parser a
-fl val opt desc = flag' val $ mconcat [long opt, help desc]
+flagParser :: a -> String -> String -> Parser a
```

Comment:
You have `lastStrOption`, and you have others you can check here - https://github.com/input-output-hk/cardano-node/blob/7c1ba8014ba62ec562dd2cd538a6f769e148c30a/cardano-config/src/Cardano/Config/CommonCLI.hs#L218-L223

Reply:
We want the various protocols to be flags though. `--praos` or `--real-pbft` is more succinct than say `--protocol praos`